Fixes likely mistake in conflict resolution of upstream backport
of a73970a8640330c4908d68ef9257fd31a4fdae9.

EIGEN_COMP_CLANG_STRICT does not exist in 3.4 branch and was not
touched in the original commit.

Gentoo-bug: https://bugs.gentoo.org/976285

--- a/Eigen/src/Core/util/ConfigureVectorization.h
+++ b/Eigen/src/Core/util/ConfigureVectorization.h
@@ -445,7 +445,7 @@
 #define EIGEN_VECTORIZE_FMA
 #endif
 
-#if defined(__F16C__) && !defined(EIGEN_GPUCC) && (!EIGEN_COMP_CLANG_STRICT || EIGEN_COMP_CLANG>=380)
+#if defined(__F16C__) && (!defined(EIGEN_GPUCC) && (!defined(EIGEN_COMP_CLANG) || EIGEN_COMP_CLANG>=380))
   // We can use the optimized fp16 to float and float to fp16 conversion routines
   #define EIGEN_HAS_FP16_C
 
